bionic/protocols.py

```diff
@@ -885,6 +885,14 @@ def validate(self, value):
             """
             raise AssertionError(oneline(message))
 
+    def _raise_validation_exception(self, descriptor_text, exception):
+        message = f"""
+        Expected a sequence. Value is invalid for
+        {self.__class__.__name__} due to {exception.__class__.__name__}:
+        {exception}. Did you mean to use @output instead of @outputs?
```

Comment:
What do you think about, rather than overriding this method, changing line 880 to produce a message more like what's on line 883? That way, `validate` is responsible for the details of what we were expecting, and `_raise_validation_expection` is just responsible for providing context like `{descriptor_text}`.

In particular, it looks like if we throw the exception on line 886, the resulting error message will say `Expected a sequence` twice, and also say `Did you mean to use @output instead of @outputs?`, which doesn't make sense for that particular error.

Reply:
Right, that's a lot cleaner - I didn't realize that bit of the error message came from 880. Fixed this in the new commit.

Reply:
Right, that's a lot cleaner - I didn't realize that bit was coming from the TypeError section in `validate` at first. I moved the clarification there like you suggested.